### PR TITLE
Add time_format argument to ExcelWriter.

### PR DESCRIPTION
**Pull request by BertrandBordage**
Currently, converting to Excel a DataFrame containing `datetime.time` instances leads to unformated float values such as `0.666667` instead of `16:00:00`. This pull request fixes this by adding a default time format and adds an argument to specify your own time format.

### CODE REVIEW

**Comment by jreback** [comment]
- needs tests

does this have an issue number?

**Comment by BertrandBordage** [comment]
@jreback OK for the tests. I guess I only need to modify [these lines](https://github.com/pydata/pandas/blob/7e3212edd642adfd916baf65691dbef5f7b03d0b/pandas/io/tests/test_excel.py#L843-880) to add a time series to `df` & `df_expected`, then add a `time_format='HH-MM-SS'` argument to the writer2 constructor?

There’s no issue related to this pull request. Do I need to create one? That seemed quite obvious to me that `time_format` has only been forgotten.

File: pandas/io/excel.py [hunk 1]
```diff
@@ -766,10 +773,13 @@ def write_cells(self, cells, sheet_name=None, startrow=0, startcol=0):
             # Apply format codes before cell.style to allow override
             if isinstance(cell.val, datetime.datetime):
                 style_kwargs.update(self._convert_to_style_kwargs({
-                        'number_format':{'format_code': self.datetime_format}}))
+                    'number_format': {'format_code': self.datetime_format}}))
             elif isinstance(cell.val, datetime.date):
                 style_kwargs.update(self._convert_to_style_kwargs({
-                        'number_format':{'format_code': self.date_format}}))
+                    'number_format': {'format_code': self.date_format}}))
+            elif isinstance(cell.val, datetime.time):
```
Comment (jreback) [inline pandas/io/excel.py]: this should actually accept time/Timedelta as well

**Comment by jreback** [comment]
ok on the issue number, use this PR number as the issue number in the release note.

Would prefer you add a new set of tests (like those though)

**Comment by jreback** [comment]
@BertrandBordage can you update?

**Comment by jreback** [comment]
closing as stale, but if you would like to reopen and fixup pls do.

**Pull request closed without merging**